src/ui/component/stepper/stepperProgress/stepperProgress.tsx

```diff
@@ -0,0 +1,51 @@
+import type { FC } from 'react'
+import { ProgressBar } from './progressBar/progressBar'
+import { getProgressBarColor, getProgressBarTransition } from '../utils'
+import { usePrevious } from '@/ui/hook/usePrevious'
+import './stepperProgress.scss'
+
+type StepId = string
+
+export type StepStatus = {
+  id: StepId
+  status: {
```

Comment:
You can keep it simple here, I guess by using a `status` enum:
```suggestion
  status: "active' | 'valid' | 'error'
```